### DISCUSSION TITLE
Sun-Panel YM-NAV 自定义css js 源码思路

### DISCUSSION THREAD

**Discussion by hslr-s** [hslr-s/sun-panel#78] [Show and tell]
欢迎大家有教程的可以来社区分享~

博客作者：YM(与末)

教程地址：[博客教程](http://blog.yumo.icu/archives/sun-panelym-nav-zi-ding-yi-cssjs-yuan-ma-si-lu)

效果：

https://github.com/hslr-s/sun-panel/assets/38825747/362062a2-16fc-4372-9c88-a38bda5a9fde

**Comment by WYithal** [comment]
又重新让GPT写了个，感觉这个代码更好，一句一个注释🤣
index.js:
```
// 定义一个函数，用来获取一言和它的出处
function getHitokoto() {
  // 创建一个XMLHttpRequest对象
  var xhr = new XMLHttpRequest();

  // 设置请求的URL和方法
  xhr.open("GET", "https://v1.hitokoto.cn/?encode=json");

  // 设置请求完成后的回调函数
  xhr.onload = function() {
    // 如果响应状态码为200，表示成功
    if (xhr.status == 200) {
      // 将响应文本转换为JS对象
      var data = JSON.parse(xhr.responseText);

      // 提取hitokoto和from参数的内容
      var hitokoto = data.hitokoto;
      var from = data.from;

      // 获取HTML代码中的元素，假设有id为hitokoto和from的元素
      var hitokotoElement = document.getElementById("hitokoto");
      var fromElement = document.getElementById("from");

      // 将内容插入到元素中
      hitokotoElement.innerHTML = "『" + hitokoto + "』";
      fromElement.innerHTML = "——「" + from + "」";
    }
  };

  // 发送请求
  xhr.send();
}

// 当页面加载完成后，执行以下代码
window.onload = function() {
  // 调用一次getHitokoto函数，获取初始的内容
  getHitokoto();

  // 设置一个定时器，每隔10分钟调用一次getHitokoto函数
  setInterval(getHitokoto, 10 * 60 * 1000);
};

```
自定义页脚(还是GPT写的🤣)：
```
<div>
    <b><i id="hitokoto"></i></b>
    <b><i id="from"></i></b>
</div>
```
然后index.css要改改，我用的上面作者发的
```
.footer {
  color: #fff;
  position: fixed;
  left: 50%;
  transform: translateX(-50%);
  bottom: 0px;
  width: 100%;
  height: auto;
  min-height: 50px;
  z-index: 9999;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  line-height: 1.8;
  pointer-events: none;
}
```

实际效果：

> 
![123](https://github.com/hslr-s/sun-panel/assets/13925273/aa5e3102-d2ed-4953-90d7-a97e0277c18b)

**Comment by docker-master** [comment]
![image](https://github.com/hslr-s/sun-panel/assets/58912611/65bfd36d-e418-4074-becd-9e51677111d1)
所以放在自定义页脚里面吗，这个效果？？？

**Comment by WYithal** [comment]
> ![image](https://private-user-images.githubusercontent.com/58912611/302107166-65bfd36d-e418-4074-becd-9e51677111d1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDcyODMyMzksIm5iZiI6MTcwNzI4MjkzOSwicGF0aCI6Ii81ODkxMjYxMS8zMDIxMDcxNjYtNjViZmQzNmQtZTQxOC00MDc0LWJlY2QtOWU1MTY3NzExMWQxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA3VDA1MTUzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThlZmM1ZmIxMTE3ZmQ0Mjc0ZjViNWM4OGE1M2E5MmJjYTk2Njk3ZWIyYzIyNzBiOWE5ODYzM2FlNDgwM2U1ZGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.O03oHNkEvoQ-MItisB9O53qREbsoxvG8W1y43-wmMko) 所以放在自定义页脚里面吗，这个效果？？？

啊？自定义页脚放HTML代码的，JS代码得放index.js里，你可以看下作者的文档
